src/app/api/inventoryLog/route.js

```diff
@@ -0,0 +1,98 @@
+// TODO: Add authentication logic
+import { NextResponse } from 'next/server'
+import { InventoryLog } from '~/api/entities'
+import { authenticateToken } from '~/app/api/Libs/auth'
+import ERROR from '~/Libs/error'
+import cleanerData from '~/app/api/Libs/cleanerData'
+import validatorFields from '~/app/api/Libs/validatorFields'
+import prisma from '~/app/api/Libs/prisma'
+
+export const POST = async request => {
+  try{
+    const hasPermission = true //authenticateToken(request)
+    const data = await request.json()
+    const isValid = validatorFields({ data, shape: InventoryLog.shape })
+    if(hasPermission && isValid){
+        //The transaction is executed first, if it fails, it will not create the inventory log
+        const target = await prisma.lot.findFirst({
+            select:{
+                id:true,
+                currentAmount: true
+            },
+            where:{
+                product:{
+                    name: data.productName
+                },
+                expirationDate: data.expirationDate
+            }
+        })
+        if(!target) return ERROR.NOT_FOUND('Lot not found')
```

Comment:
```suggestion
        if(!target) return ERROR.NOT_FOUND()
```

El objeto error no recibe mensajes, ya hay mensajes genericos precargados y si necesitas otro en especificico puedes modificar ERROR para tus necesidades